packages/styled-base/src/utils.js

```diff
@@ -1,9 +1,31 @@
 // @flow
 import * as React from 'react'
+import type { ElementType } from 'react'
 import isPropValid from '@emotion/is-prop-valid'
 
 export type Interpolations = Array<any>
 
+export type StyledOptions = {
+  label?: string,
+  shouldForwardProp?: string => boolean,
+  target?: string
+}
+
+export type StyledComponent = {
+  (Interpolations): React$Node,
+  withComponent: (
+    nextTag: ElementType,
+    nextOptions?: StyledOptions
+  ) => StyledComponent,
+  displayName: string,
+  defaultProps: any,
+  __emotion_real: StyledComponent,
```

Comment:
IMHO those `__emotion` properties should be hidden from the user. Instead of adding them here to the declared type we can just cast Styled to any when assigning those.

Reply:
I’d argue that Flow types are not only for the consumers, they should help also the Emotion maintainers to write tipe safe code, so “any” should never be used

Reply:
those are literally used in a single file and are not supposed to be used outside of this, so I would say it's better to hide them from the outside world to avoid misusage

Reply:
Now we expose a type without them, but internally we keep still type check them, so we get the best of both worlds.